tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +302,110 @@ def guess_realert_every(job):
+        'tags',
+        'ttl',
+        'sensu_host',
+        'component',
+        'description',
+    ])
+
+    include, exclude = {}, {}
+    for k, v in config.items():
+        (include if k in sensu_args else exclude)[k] = v
+    return include, exclude
+
+
+def sort_runs_by_interval(job_content, interval='day'):
+    """ Sorts a job's runs by a time interval (year, month, day, hour, minute,
+    or second), according to a job run's run time.
+    """
+    interval_formats = {
+        'year': '%Y',
+        'month': '%Y.%m',
+        'day': '%Y.%m.%d',
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
-    if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
 
-    status = job["status"]
-    if status == "disabled":
-        kwargs["output"] = "OK: {} is disabled and won't be checked.".format(
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        # Sort runs by shortdate
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            shortdate = time.strftime(
+                interval_formats[interval],
+                _timestamp_to_timeobj(run['run_time']),
+            )
+            run_buckets[shortdate].append(run)
+    return dict(run_buckets)
+
+
+def compute_check_result_for_job(client, job):
+    kwargs = m(
+        name="check_tron_job.{}".format(job['name']),
+        source="tron",
+    )
+    if 'realert_every' not in kwargs:
+        kwargs = kwargs.set('realert_every', guess_realert_every(job))
+    kwargs = kwargs.set('check_every', "{}s".format(_run_interval))
+    sensu_kwargs, nonsensu_kwargs = filter_monitoring_config(job['monitoring'])
+    kwargs = kwargs.update(sensu_kwargs)
+
+    kwargs_list = []
+    if job["status"] == "disabled":
+        kwargs = kwargs.set('output', "OK: {} is disabled and won't be checked.".format(
             job['name'],
-        )
-        kwargs["status"] = 0
-        log.info(kwargs["output"])
-        return kwargs
+        ))
+        kwargs = kwargs.set('status', 0)
+        kwargs_list.append(kwargs)
     else:
         # The job is not disabled, therefore we have to look at its run history
         url_index = client.index()
         tron_id = get_object_type_from_identifier(url_index, job["name"])
-        job_content = client.job(
+        job_content = pmap(client.job(
             tron_id.url,
             count=20,
             include_action_runs=True,
-        )
-        results = compute_check_result_for_job_runs(
-            job=job,
-            job_content=job_content,
-            client=client,
-        )
-        kwargs.update(results)
-        log.info(kwargs["output"].split("\n")[0])
-        return kwargs
+        ))
+
+        if _are_runs_precious(nonsensu_kwargs):
```

Comment:
Hehe. Using this as a dictionary "as if" it was monitoring.yaml feels a bit skecth, but I don't have any super great alternatives to recommend?

Reply:
On @Rob-Johnson, I'm removing `_are_runs_precious`. Also, I don't have to use `nonsensu_kwargs`, I can still use `job['monitoring']`.